sympy/stats/stochastic_process_types.py

```diff
@@ -547,22 +602,24 @@ def expectation(self, expr, condition=None, evaluate=True, **kwargs):
         if check:
             return Expectation(expr, condition)
 
-        if isinstance(expr, RandomIndexedSymbol):
-            if isinstance(condition, Eq):
-                # handle queries similar to E(X[i], Eq(X[i-m], <some-state>))
-                lhsg, rhsg = condition.lhs, condition.rhs
-                if not isinstance(lhsg, RandomIndexedSymbol):
-                    lhsg, rhsg = (rhsg, lhsg)
-                if rhsg not in self.state_space:
-                    raise ValueError("%s state is not in the state space."%(rhsg))
-                if expr.key < lhsg.key:
-                    raise ValueError("Incorrect given condition is given, expectation "
-                      "time %s < time %s"%(expr.key, lhsg.key))
-                cond = condition & TransitionMatrixOf(self, trans_probs) & \
-                        StochasticStateSpaceOf(self, state_space)
-                s = Dummy('s')
-                func = Lambda(s, self.probability(Eq(expr, s), cond)*s)
-                return Sum(func(s), (s, state_space.inf, state_space.sup)).doit()
+        rvs = random_symbols(expr)
+        if isinstance(expr, Expr) and isinstance(condition, Eq) \
+            and len(rvs) == 1:
+            # handle queries similar to E(f(X[i]), Eq(X[i-m], <some-state>))
+            rv = list(rvs)[0]
+            lhsg, rhsg = condition.lhs, condition.rhs
+            if not isinstance(lhsg, RandomIndexedSymbol):
+                lhsg, rhsg = (rhsg, lhsg)
+            if rhsg not in self.state_space:
+                raise ValueError("%s state is not in the state space."%(rhsg))
+            if rv.key < rv.key:
```

Comment:
Why would these be different?

Reply:
Ah! made a mistake, actually one should be rv from expr and the other one should be from condition. It's a typo.  Thanks for notifying.